Frontend-unit-tests-guide.md

```diff
@@ -413,59 +488,56 @@ it('should wait for 10 seconds to call console.log', function() {
 ```
 
 **Good code:**
-```
+
+```js
 it('should wait for 10 seconds to call console.log', function() {
   spyOn(console, 'log');
   $flushPendingTasks();
   expect(console.log).toHaveBeenCalled();
 });
 ```
 
-#### Mocking with `$q` API in AngularJS  
-When mocking a promise in AngularJS, you might use `$q` API. In these cases, you must use `$scope.$apply()` or `$scope.$digest` because it forcibly `$q` promises to be resolved through a Javascript digest. Here are some examples using [$apply](https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/pages/email-dashboard-pages/email-dashboard-page.controller.spec.ts#L101-L108) and [$digest](https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/pages/exploration-editor-page/services/exploration-states.service.spec.ts#L209-L221).
+#### Mocking with `$q` API in AngularJS
 
-### When should the upgraded services be imported in the test file?  
-One of the active projects in Oppia is the Angular2+ migration. By now, the AngularJS services are being migrated and it’s still being used in downgrade files. When testing AngularJS files which uses an Angular2+ as a dependency, you must use the beforeEach call below: 
-Let us assume that the test requires MyExampleService which is an Angular2+ service.
+When mocking a promise in AngularJS, you might use the `$q` API. In these cases, you must use `$scope.$apply()` or `$scope.$digest` because it forces `$q` promises to be resolved through a Javascript digest. Here are some examples using [$apply](https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/pages/email-dashboard-pages/email-dashboard-page.controller.spec.ts#L101-L108) and [$digest](https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/pages/exploration-editor-page/services/exploration-states.service.spec.ts#L209-L221).
```

Comment:
```suggestion
When mocking a promise in AngularJS, you might use the `$q` API. In these cases, you must use `$scope.$apply()` or `$scope.$digest` because they force `$q` promises to be resolved through a Javascript digest. Here are some examples using [$apply](https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/pages/email-dashboard-pages/email-dashboard-page.controller.spec.ts#L101-L108) and [$digest](https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/pages/exploration-editor-page/services/exploration-states.service.spec.ts#L209-L221).
```